File: src/command_executor.py
from datetime import datetime
import locale

try:
    locale.setlocale(locale.LC_TIME, 'pt_BR.UTF-8')
except locale.Error:
    print("Localidade 'pt_BR.UTF-8' não encontrada. O nome do mês pode ficar em inglês.")

def execute_local_command(command: str) -> str | None:

    command = command.lower()

    if "que horas são" in command or "me diga as horas" in command:
        now = datetime.now()
        return f"Agora são {now.hour} horas e {now.minute} minutos."

    if "que dia é hoje" in command or "qual a data de hoje" in command:
        now = datetime.now()
        return now.strftime("Hoje é dia %d de %B de %Y.")
        
    return None

File: backend/src/commandExecutor.js

function executeLocalCommand(command) {
    const lowerCaseCommand = command.toLowerCase();

    if (lowerCaseCommand.includes("que horas são") || lowerCaseCommand.includes("me diga as horas")) {
        const now = new Date();

        const timeString = now.toLocaleTimeString('pt-BR', { hour: '2-digit', minute: '2-digit' });
        return `Agora são ${timeString}.`;
    }

    if (lowerCaseCommand.includes("que dia é hoje") || lowerCaseCommand.includes("qual a data de hoje")) {
        const now = new Date();

        const dateString = now.toLocaleDateString('pt-BR', { day: 'numeric', month: 'long', year: 'numeric' });
        return `Hoje é ${dateString}.`;
    }


    return null;
}


module.exports = { executeLocalCommand };